server/cli/src/cli.rs

```diff
@@ -396,6 +405,102 @@ async fn main() -> anyhow::Result<()> {
             info!("Refresh data result: {:#?}", result);
         }
         Action::SignPlugin { path, key, cert } => sign_plugin(&path, &key, &cert)?,
+        Action::UpsertStandardReports { name } => {
+            let connection_manager = get_storage_connection_manager(&settings.database);
+            let con = connection_manager.connection()?;
+            let base_reports_dir = "./reports";
+
+            let report_names = fs::read_dir(base_reports_dir)?
+                .map(|res| res.map(|e| e.path().into_os_string().into_string().unwrap()))
+                .collect::<Result<Vec<String>, std::io::Error>>()?;
+
+            let report_names = match &name {
+                Some(name) => report_names
+                    .into_iter()
+                    .filter(|report| report == name)
+                    .collect(),
+                None => report_names,
+            };
+
+            for name_dir in report_names {
+                let report_versions = fs::read_dir(&name_dir)?
+                    .map(|res| res.map(|e| e.path().into_os_string().into_string().unwrap()))
+                    .collect::<Result<Vec<String>, std::io::Error>>()?;
+
+                let (_, name) = name_dir.rsplit_once('/').unwrap();
+
+                for version_dir in report_versions {
+                    let (_, version) = version_dir.rsplit_once('/').unwrap();
+
+                    let args = BuildArgs {
+                        dir: format!("{version_dir}/src"),
+                        output: format!("{version_dir}/generated/{name}.json").into(),
+                        template: "template.html".to_string(),
+                        header: None,
+                        footer: None,
+                        query_gql: Some("query.graphql".to_string()),
+                        query_default: None,
+                        query_sql: None,
+                    };
+
+                    let _ = build(args);
+
+                    let arguments_path = format!("{version_dir}/argument_schemas/arguments.json");
+                    let arguments_ui_path =
+                        format!("{version_dir}/argument_schemas/arguments_ui.json");
+
+                    let manifest_file = fs::File::open(format!("{version_dir}/manifest.json"))
+                        .expect("file should open read only");
+
+                    let manifest: Manifest = serde_json::from_reader(manifest_file)
+                        .expect("manifest json not formatted");
+
+                    let id = format!("{name}_{version}");
+                    let report_path = format!("{version_dir}/generated/{name}.json");
+                    let context = manifest.context;
+                    let report_name = manifest.name;
+                    let is_custom = manifest.is_custom;
+
+                    let sub_context = manifest.sub_context;
+                    let code = manifest.code;
+
+                    let filter = ReportFilter::new().id(EqualFilter::equal_to(&id));
```

Comment:
I think Andrei's idea was to use the `code` to identify the matching report.
I think this is probably fine too, as long as there's no duplicate names...

Reply:
Yes thank you James, I had just picked up on this also after adding this PR